stdlib/3/typing.pyi

```diff
@@ -520,7 +520,10 @@ class NamedTuple(tuple):
     @classmethod
     def _make(cls, iterable: Iterable[Any]) -> NamedTuple: ...
 
-    def _asdict(self) -> dict: ...
+    if sys.version_info >= (3, 1):
+        def _asdict(self) -> collections.OrderedDict: ...
```

Comment:
Can you add type arguments (probably `[str, Any]`)?

Reply:
Added on the Dict block but for OrderedDict it crashes. Is there any workaround to allow typed OrderedDict?

Reply:
What do you mean by crashes? What errors do you get?

Reply:
I think this would cause an exception as in the example below:
```
(.venv3) luka-mbp:typeshed luka$ cat test.py
from collections import OrderedDict

def new_dict() -> OrderedDict[str, str]:
    return OrderedDict()

print(new_dict())
(.venv3) luka-mbp:typeshed luka$ python3 test.py
Traceback (most recent call last):
  File "test.py", line 3, in <module>
    def new_dict() -> OrderedDict[str, str]:
TypeError: 'type' object is not subscriptable
```

Reply:
@JelleZijlstra how do you want to proceed?

Reply:
I had written a comment here but I guess I forgot to post it.

It's true that OrderedDict with type arguments doesn't work at runtime, but they should work fine in stubs, because OrderedDict is generic in the stub.

Reply:
Nice, didn't know that 👍